osquery/events/darwin/es_utils.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <iomanip>
+#include <osquery/core/flags.h>
+#include <osquery/events/darwin/endpointsecurity.h>
+#include <osquery/logger/logger.h>
+#include <pwd.h>
+
+namespace osquery {
+
+FLAG(bool,
+     disable_endpointsecurity,
+     true,
+     "Disable receiving events from the EndpointSecurity subsystem");
+
+FLAG(bool,
+     disable_endpointsecurity_fim,
+     true,
+     "Disable file events from the EndpointSecurity subsystem");
+
+FLAG(string,
+     es_fim_mute_path_literal,
+     "",
+     "Comma delimited list of path literals to be muted for FIM");
+
+FLAG(string,
+     es_fim_mute_path_prefix,
+     "",
+     "Comma delimited list of path prefxes to be muted for FIM");
+
+std::string getEsNewClientErrorMessage(const es_new_client_result_t r) {
+  switch (r) {
+  case ES_NEW_CLIENT_RESULT_ERR_INTERNAL:
+    return "Internal Error";
+  case ES_NEW_CLIENT_RESULT_ERR_INVALID_ARGUMENT:
+    return "Invalid Argument";
+  case ES_NEW_CLIENT_RESULT_ERR_NOT_ENTITLED:
+    return "EndpointSecurity client lacks entitlement";
+  case ES_NEW_CLIENT_RESULT_ERR_NOT_PERMITTED:
+    return "EndpointSecurity client lacks user TCC permissions";
+  case ES_NEW_CLIENT_RESULT_ERR_NOT_PRIVILEGED:
+    return "EndpointSecurity client is not running as root";
+  case ES_NEW_CLIENT_RESULT_ERR_TOO_MANY_CLIENTS:
+    return "Too many EndpointSecurity clients running on the system";
+  default:
+    return "EndpointSecurity: Unknown Error";
+  }
+}
+
+std::string getPath(const es_process_t* p) {
+  return p->executable->path.length > 0 ? p->executable->path.data : "";
+}
+
+std::string getSigningId(const es_process_t* p) {
+  return p->signing_id.length > 0 && p->signing_id.data != nullptr
+             ? p->signing_id.data
+             : "";
+}
+
+std::string getTeamId(const es_process_t* p) {
+  return p->team_id.length > 0 && p->team_id.data != nullptr ? p->team_id.data
+                                                             : "";
+}
+
+std::string getStringFromToken(es_string_token_t* t) {
```

Comment:
Is this non-const version of the function necessary?

Reply:
Ah, good catch! We only use the non-const version (since that's what the ES API needs), removed the dead code that I was trying.